quarkchain/evm/messages.py

```diff
@@ -137,6 +136,53 @@ def validate_transaction(state, tx):
     if not tx.sender:  # sender is set and validated on Transaction initialization
         raise UnsignedTransaction(tx)
 
+    if tx.version == 2:
+        # When tx.version == 2 (EIP155 tx), check
+        # 0. EIP155_SIGNER enable
+        # 1. tx.v == tx.network_id * 2 + 35 (+ 1)
+        # 2. gas_token_id & transfer_token_id should equal to default_token_id (like qkc)
+        # 3. tx.from_chain_id, tx.to_chain_id should match with chain_config.ETH_CHAIN_ID
```

Comment:
Shouldn't tx.network_id match chain_config.ETH_CHAIN_ID?

So it should be

3, tx.from_chain_id == tx.to_chain_id
4, tx.network_id == chain_config.ETH_CHAIN_ID, where chain_config is derived from tx.from_chain_id